modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/portlet/FDSViewsPortlet.java

```diff
@@ -589,28 +589,190 @@ private synchronized void _generate(
 			_objectDefinitionLocalService.fetchObjectDefinition(
 				companyId, "FDSEntry");
 
-		if (fdsEntryObjectDefinition != null) {
-			return;
-		}
+		ObjectDefinition fdsDateFilterObjectDefinition =
+			_objectDefinitionLocalService.addSystemObjectDefinition(
+				"FDSDateFilter", userId, 0, "FDSDateFilter", "FDSDateFilter",
+				false, LocalizedMapUtil.getLocalizedMap("FDS Date Filter"),
+				true, "FDSDateFilter", null, null, null, null,
+				LocalizedMapUtil.getLocalizedMap("FDS Date Filters"),
+				ObjectDefinitionConstants.SCOPE_COMPANY, null, 1,
+				WorkflowConstants.STATUS_DRAFT,
+				Arrays.asList(
+					ObjectFieldUtil.createObjectField(
+						ObjectFieldConstants.BUSINESS_TYPE_DATE,
+						ObjectFieldConstants.DB_TYPE_DATE, true, false, null,
+						_language.get(locale, "to"), "to", false),
+					ObjectFieldUtil.createObjectField(
+						ObjectFieldConstants.BUSINESS_TYPE_DATE,
+						ObjectFieldConstants.DB_TYPE_DATE, true, false, null,
+						_language.get(locale, "from"), "from", false),
+					ObjectFieldUtil.createObjectField(
+						ObjectFieldConstants.BUSINESS_TYPE_TEXT,
+						ObjectFieldConstants.DB_TYPE_STRING, true, false, null,
+						_language.get(locale, "field-name"), "fieldName", true),
+					ObjectFieldUtil.createObjectField(
+						ObjectFieldConstants.BUSINESS_TYPE_TEXT,
+						ObjectFieldConstants.DB_TYPE_STRING, true, false, null,
+						_language.get(locale, "type"), "type", false)));
+
+		_enableLocalization(fdsDateFilterObjectDefinition);
 
-		fdsEntryObjectDefinition = _createFDSEntryObjectDefinition(
-			locale, userId);
+		_addLocalizedCustomObjectField(
+			_language.get(locale, "name"), "name",
+			fdsDateFilterObjectDefinition, userId);
+
+		_objectDefinitionLocalService.publishSystemObjectDefinition(
+			userId, fdsDateFilterObjectDefinition.getObjectDefinitionId());
 
 		ObjectDefinition fdsViewObjectDefinition =
 			_createFDSViewObjectDefinition(
 				fdsEntryObjectDefinition, locale, userId);
 
-		_createFDSActionObjectDefintion(
-			fdsViewObjectDefinition, locale, userId);
-		_createFDSClientExtensionFilterObjectDefintion(
-			fdsViewObjectDefinition, locale, userId);
-		_createFDSDateFilterObjectDefinition(
-			fdsViewObjectDefinition, locale, userId);
-		_createFDSDynamicFilterObjectDefintion(
-			fdsViewObjectDefinition, locale, userId);
-		_createFDSFieldObjectDefinition(
-			fdsViewObjectDefinition, locale, userId);
-		_createFDSSortObjectDefinition(fdsViewObjectDefinition, locale, userId);
+		ObjectDefinition fdsDynamicFilterObjectDefinition =
```

Comment:
we'll need to move all this to the corresponding methods `_create*****ObjectDefintion()`

Reply:
Ouch